Use newer config.(guess|sub) to support more CPU architectures (e.g. aarch64)

martin-g opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [X] Used a fork of the feedstock to propose changes
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

martin-g committed:
Use newer config.(guess|sub) to support more CPU architectures (e.g. aarch64)

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Fix lint errors

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
@conda-forge-admin, please rerender

martin-g commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.1.2, conda-smithy 3.33.0, and conda-forge-pinning 2024.03.22.08.38.34

martin-g committed:
Fix license_file value

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.11.2, conda-smithy 3.45.1, and conda-forge-pinning 2025.01.02.20.43.49

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ This recipe is using a compiler, which now requires adding a build dependence on `{{ stdlib("c") }}` as well. Note that this rule applies to each output of the recipe using a compiler. For further details, please see https://github.com/conda-forge/conda-forge.github.io/issues/2102.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/12595751088](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12595751088). Examine the logs at this URL for more detail.</sub>

martin-g commented on recipe/meta.yaml:
```diff
@@ -11,22 +11,25 @@ source:
   build:
     - make
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
```
```suggestion
    - {{ compiler('cxx') }}
    - {{ stdlib('c') }}
```

martin-g committed:
Add stdlib(c)

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

martin-g committed:
Build for aarch64/ppc64le/arm64

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.11.2, conda-smithy 3.45.1, and conda-forge-pinning 2025.01.02.20.43.49

martin-g committed:
Do not cross-compile. `./configure`'s tests do not work with cross-compiling

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.11.2, conda-smithy 3.45.1, and conda-forge-pinning 2025.01.02.20.43.49

martin-g committed:
Move `wget` to `build` reqs

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g committed:
Try with emulation on Azure for aarch64/ppc64le

martin-g commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.11.2, conda-smithy 3.45.1, and conda-forge-pinning 2025.01.03.13.11.43

martin-g committed:
Use emulation for osx-arm64 too

martin-g commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/12599428606](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12599428606). Examine the logs at this URL for more detail.</sub>

martin-g commented:
With https://github.com/conda-forge/ossuuid-feedstock/pull/9/commits/ab5eb75cd052c157702b19bca320116172448f25 I tried to use emulation for OSX ARM64 but it seems this does not work.

martin-g commented on conda-forge.yml:
```diff
@@ -1,9 +1,11 @@
 conda_build:
   pkg_format: '2'
+provider:
+  osx_arm64: azure
```
```suggestion
```

martin-g committed:
Do not try to build osx-arm64. The configure tests do not pass when cross-compiling

martin-g commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 24.11.2, conda-smithy 3.45.1, and conda-forge-pinning 2025.01.03.14.12.46

martin-g commented:
@conda-forge-admin, please ping team

mfansler commented on recipe/build.sh:
```diff
@@ -1,6 +1,13 @@
+
+# update config.guess and config.sub to newer versions which support more CPU architectures
+rm -f config.sub config.guess
+wget "http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD" -O config.guess
+wget "http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD" -O config.sub
```
Better form would be to require `gnuconfig` and copy from there to replace the build source. There's an example in https://github.com/conda-forge/r-fs-feedstock/pull/27.

martin-g replied:
Thank you for the suggestion, @mfansler !
Done with https://github.com/conda-forge/ossuuid-feedstock/pull/9/commits/42d295207b5a3e931b04c446fd7617a56798ec2c !

martin-g committed:
Use gnuconfig as a build requirement

And copy config.sub/.guess from it, to be able to support linux-aarch64
and osx-arm64

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

mfansler reviewed (approved):
Not a maintainer here, but looks good to me. Thanks for refreshing this ancient feedstock!

martin-g commented:
@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like there was nothing to do.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/12744399445](https://github.com/conda-forge/conda-forge-webservices/actions/runs/12744399445). Examine the logs at this URL for more detail.</sub>

martin-g commented:
@conda-forge/core Could you please help with the review and merge of this PR ? Thank you!
